src/utils/curriculum/openapi.ts

```diff
@@ -3,10 +3,28 @@ import { z } from "zod";
 import { Unit } from "./types";
 import { CurriculumSelectionSlugs } from "./slugs";
 
-const unitSchema = z.object({
+const categorySchema = z.object({
+  categoryTitle: z.string(),
+});
+
+const threadSchema = z.object({
+  title: z.string(),
+  slug: z.string(),
+  order: z.number(),
+});
+
+const unitOptionSchema = z.object({
   unitTitle: z.string(),
   unitSlug: z.string(),
-  order: z.number(),
+});
+
+const unitSchema = z.object({
+  unitTitle: z.string(),
+  unitSlug: z.string().default("UNKNOWN"),
```

Comment:
This is sometime missing, so set to `UNKNOWN` if not present

Reply:
When it's missing, it means that it's a unit with optionality - the API intentionally doesn't provide this, because looking up the unit with the "parent" slug returns nothing/errors.

Reply:
I think it should have a slug, leave it with me I'll double check in the data.